internal/handler/task.go

```diff
@@ -77,30 +95,76 @@ func (h *Handler) ModCmd() *cobra.Command {
 		Use:   "mod",
 		Short: "Modify a task by ID",
 		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Println("modifies task")
+			ti, err := parseTask(args, false)
+			if err != nil {
+				log.Fatal(err)
+			}
+			t, err := h.service.GetTask(*ti.id)
+			if err != nil {
+				return
+			}
+			//check if task exists, because of nil pointer dereference
+			if t == nil {
+				fmt.Printf("Task %d not found.\n", *ti.id)
+				return
+			}
+			fmt.Printf("Task %d '%s' has been updated:\n", t.ID, t.Desc)
+			if ti.desc != nil {
+				fmt.Printf("  - Description updated to '%s'\n", *ti.desc)
+				t.Desc = *ti.desc
+			}
+			if ti.priority != nil {
+				fmt.Printf("  - Priority updated from %d to %d\n", t.Priority, *ti.priority)
+				t.Priority = *ti.priority
+			}
+			if ti.addTags != nil {
+				for _, tg := range ti.addTags {
+					fmt.Printf("  - Tag added: %s\n", tg)
+				}
+				t.Tags = append(t.Tags, ti.addTags...)
+			}
+			if ti.startAt != nil {
+				t.StartAt = ti.startAt
+			}
+			if ti.endAt != nil {
+				t.EndAt = ti.endAt
+			}
+			if ti.startAt != nil {
+				fmt.Printf("  - Time updated to ")
+				if t.StartAt != nil && t.EndAt != nil {
+					fmt.Printf("%s %s - %s\n", t.StartAt.Format("Mon, 02 Jan 2006"), t.StartAt.Format(time.Kitchen), t.EndAt.Format(time.Kitchen))
+				} else {
+					fmt.Printf("%s %s\n", t.StartAt.Format("Mon, 02 Jan 2006"), t.StartAt.Format(time.Kitchen))
+				}
+			}
+			fmt.Println("Update complete. 1 task modified.")
 		},
 	}
 	return editCmd
 }
 
-func (h *Handler) ComCmd() *cobra.Command {
+func (h *Handler) NoteCmd() *cobra.Command {
 	comCmd := &cobra.Command{
-		Use:   "com",
-		Short: "Comment a task by ID",
+		Use:   "note",
+		Short: "Notes a task by ID",
+		Args:  cobra.ExactArgs(2),
 		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) != 2 {
-				log.Fatal(errors.New("argument mismatch for comment command"))
-			}
-			id, err := strconv.Atoi(args[0])
+			id, n, err := parseNote(args)
 			if err != nil {
 				log.Fatal(err)
 			}
-			task, err := h.service.GetTask(id)
+			t, err := h.service.GetTask(id)
 			if err != nil {
+				log.Fatal(err)
+			}
+			//check if task exists, because of nil pointer dereference
+			if t == nil {
```

Comment:
We have to check, if task is not nil, because it will always throw "panic: runtime error: invalid memory address or nil pointer dereference"